### ISSUE TITLE
Building `create-pkg`

### ISSUE THREAD

**Issue by darcyclarke** [nodejs/package-maintenance#404]
> **EDIT by @boneskull: This issue has been moved to pkgjs/create#9; please comment there**

Recently, I was able to transfer the ownership of the `create-pkg` package over to the `@pkgjs` org on **npmjs.com**. The hope here is that this namespace/package can serve as tooling to scaffold new projects with whatever defaults this group wants to define & be unblocked by `npm`-specific defaults in `npm init`.

@wesleytodd @ruyadorno & I got together the other day to discuss this a bit more & here's some of the notes from that discussion:

> **Note:** We'd love input & to have a broader discussion about how/what we could build

### Goals:
1. Provide a tool that promotes sane package defaults (defined by PMWG)
1. Provide tools to extend defaults per community/team

#### Example Usage:

```bash
$ npm init pkg
$ npx create-pkg
```

```js
const pkg = require('create-pkg')
import pkg from 'create-pkg'
```

### Questions:

1. ***What do we want the scaffold output to include?***
    * `package.json`
        * `name`
        * `description`
        * `homepage`
        * `author`
        * `license`
        * `funding`
        * `version`
        * `type`
        * `main`
        * `scripts`
            * `"prepublish": "..."`* (contentious)
            * `"test": "..."`* (contentious)
        * `exports`
    * `index.js`
    * `LICENSE`
    * `CONTRIBUTING.md`
    * `CODE_OF_CONDUCT.md`
    * `README.md`
    * `FUNDING.yml`* (contentious)
    * `.npmrc`* (contentious)
    * `.gitignore`* (contentious)
        * Contents: `/node_modules`
    * `.github/ISSUE_TEMPLATES/*`* (contentious)
1. ***What API surface do we want to expose to both end users and folks who want to compose the base generator with their own custom opinions?***
    * Interfaces: CLI w/ Interactive/Prompts & a JS API

### Next Steps:
* Get some general consensus of the approach we'd like to take building this & the defaults we'd want
* Update `@pkgjs/support` to be a consumer of this tool or vice-versa

### References & Prior Art:
* Github gitignore: https://github.com/github/gitignore/blob/master/Node.gitignore
* Collection user input: https://github.com/wesleytodd/opta
* Yeoman
* Plop
* `npm init x` (ie. `$ npx create-x`)
* And probably many many more scaffolding tools/generators...

**Comment by mhdawson** [comment]
In meeting was mentioned that

CONTRIBUTORS.md

should be 

CONTRIBUTING.md

**Comment by Christian24** [comment]
I hope this is the correct place to post this: Since npm v7 supports workspaces, it would be helpful to have a variant of this potentially be able to setup workspaces/monorepositories too. Maybe some inspiration can be drawn from https://github.com/lerna/lerna/tree/master/commands/init#readme. I probably cannot attend tomorrow, but since I just heard about this at the npm RFC meeting, I thought I would drop it here.

**Comment by wesleytodd** [comment]
While I am not sure that would be a top priority for the first iteration, I completely agree that it should be a feature of this tool.

**Comment by ruyadorno** [comment]
@wesleytodd please remember to also add me to the org 😊

**Comment by thescientist13** [comment]
Count me in to help on next steps @wesleytodd !  👋

**Comment by rxmarbles** [comment]
@wesleytodd ditto for me as well

**Comment by rodion-arr** [comment]
Will be happy to help here

**Comment by wesleytodd** [comment]
This one is waiting on me, I have I think all of the code on my machine, just need to run through it and push.  Sorry for the delay, I will try to get to it this week.

**Comment by boneskull** [comment]
I'm looking for some more context on this.  @darcyclarke describes what the package is intended to do, but does not describe what lead the team to want to build this.

**Comment by boneskull** [comment]
For `scripts.test`, I like `"echo \"Error: no test specified\" && exit 1"` :smile:

**Comment by mhdawson** [comment]
@wesleytodd any chance you could just push what you have, warts and all ?

**Comment by wesleytodd** [comment]
After pushing I noticed a few things wrong with what I pushed, but here is the start: https://github.com/pkgjs/create

**Comment by boneskull** [comment]
@boneskull move this list into pkgjs/create repo

**Comment by boneskull** [comment]
@wesleytodd your friendly 🙂😁😄😜 reminder to move `create-package-json` etc. into pkgjs org

**Comment by boneskull** [comment]
Closing in lieu of pkgjs/create#9

**Comment by boneskull** [comment]
Maybe we should keep this open as a "tracking" issue for `package-maintenance-agenda`?  IDK.  It should still probably stay on the agenda... new issue?
